### ISSUE TITLE
Wrong errors reported: expected Expr

### ISSUE THREAD

**Issue by yipu3** [rust-lang/rust-analyzer#17424]
**rust-analyzer version**: rust-analyzer version: 0.3.1995-standalone (c86d6230f 2024-06-10) [/Users/Philip/.vscode/extensions/rust-lang.rust-analyzer-0.3.1995-darwin-x64/server/rust-analyzer]

**rustc version**: rustc 1.58.1 (db9d1b20b 2022-01-20)

**editor or extension**: VSCode, v0.3.1995

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**repository link (if public, optional)**: https://github.com/yipu3/algorithm-practice/tree/main/leetcode/502/rust

**code snippet to reproduce**: 
```rust
use std::collections::BinaryHeap;

#[derive(Debug)]
struct Project {
    profit: i32,
    capital: i32,
}

pub fn find_maximized_capital(k: i32, w: i32, profits: Vec<i32>, capital: Vec<i32>) -> i32 {
    let mut projects = Vec::new();
    for i in 0..profits.len() {
        projects.push(Project {
            profit: profits[i],
            capital: capital[i],
        });
    }
    projects.sort_by(|a, b| a.capital.cmp(&b.capital));
    // println!("{:?}", projects);

    let mut w = w;
    let mut h = BinaryHeap::new();
    let mut c = 0;
    let mut i = 0;

    loop {
        while i < projects.len() && projects[i].capital <= w {
            h.push(projects[i].profit);
            i += 1;
        }
        if h.peek().is_none() {
            break;
        }
        w += h.pop().unwrap();
        c += 1;
        if c == k {
            break;
        }
    }

    w
}

fn main() {
    assert_eq!(
        find_maximized_capital(2, 0, vec![1, 2, 3], vec![0, 1, 1]),
        4
    );
}

```
<img width="588" alt="Screen Shot 2024-06-15 at 08 28 58" src="https://github.com/rust-lang/rust-analyzer/assets/5365604/8b8666dd-bcfe-4b16-9936-5b00c12cf2cd">

<img width="649" alt="Screen Shot 2024-06-15 at 08 31 24" src="https://github.com/rust-lang/rust-analyzer/assets/5365604/8a3af6f4-588b-415e-8a8e-7b8613a6e1f2">

**Comment by Veykril** [comment]
Your rustc version is very old compared to the rust-analyzer releases so its likely to fail because of that given the error is within a standard library macro. Either upgrade your rust toolchain or downgrade your rust-analyzer, we generally dont support more than the ~2 latest stable releases for a given rust-analyzer version

**Comment by TroyKomodo** [comment]
I also have this issue on the latest version of rust / nightly release of rust-analyzer.

![image](https://github.com/user-attachments/assets/a8570686-1350-4172-9971-88df3b16a7d8)

**Comment by TroyKomodo** [comment]
Why did you close the issue as complete it is still currently an issue. you can unsubscribe from the issue mailing list here 
<img width="1053" alt="image" src="https://github.com/user-attachments/assets/50b4d4d5-99ce-45ad-85e9-6fff4501bbb5">

**Comment by yipu3** [comment]
> Why did you close the issue as complete it is still currently an issue. you can unsubscribe from the issue mailing list here <img alt="image" width="1053" src="https://private-user-images.githubusercontent.com/49777269/365036226-50b4d4d5-99ce-45ad-85e9-6fff4501bbb5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDIzOTEsIm5iZiI6MTcyNTYwMjA5MSwicGF0aCI6Ii80OTc3NzI2OS8zNjUwMzYyMjYtNTBiNGQ0ZDUtOTljZS00NWFkLTg1ZTktNmZmZjQ1MDFiYmI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDA1NTQ1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5MGZiMjFlOGM4YTE4MDYwYWM1MzNlM2EyNDAzNDA3NzY4OGZjYjMyYTRlZjk0ZGI4NDIyZjc4NDVmOTk2MGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ccIAnVaPYCxXr2TnhxusicdVBuH9gxlgemkpyFPQwi0">

My problem has been solved by updating the rust version, suggested by the comment of Veykril. You can open a new issue providing details of your environment. That's more convinient.
